spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.lock;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.springframework.integration.support.locks.DefaultLockRegistry;
+import org.springframework.integration.support.locks.ExpirableLockRegistry;
+import org.springframework.integration.support.locks.LockRegistry;
+import org.springframework.integration.util.UUIDConverter;
+import org.springframework.util.Assert;
+
+/**
+ *
+ * A {@link LockRegistry} using a shared database to co-ordinate the locks. Provides the
+ * same semantics as the {@link DefaultLockRegistry}, but the locks taken will be global,
+ * as long as the underlying database supports the "serializable" isolation level in its
+ * transactions.
+ *
+ * @author Dave Syer
+ *
+ */
+public class JdbcLockRegistry implements ExpirableLockRegistry {
+
+	private final Map<String, JdbcLock> locks = new HashMap<String, JdbcLock>();
+
+	private JdbcClient client;
```

Comment:
With different clients we look like working. We really are rejected by the:
```
[The statement was aborted because it would have caused a duplicate key value in a unique or primary key constraint
```
until the first lock is unclocked.
But that is for `lockInterruptibly()`. I think the `lock()` has to be fixed with the `while (!locked) {` as well.